### PR TITLE
Reverted Entity __eq__ comparison on all fields, created compare to m…

### LINKED ISSUE #390: Make __eq__ compare all properties and create method "compare" to allow more granular comparisons
related to #388

### PR DESCRIPTION
**Pull request by renzon**
…ore granular comparisons

close #390

### CODE REVIEW

**Comment by renzon** [comment]
@abalakh @rochacbruno I think this make comparison more explicit. I called the method `compare` but let me know if you guys have a better name. As you may know, naming is the most difficult on Software Engineering ;)

File: nailgun/entity_mixins.py [hunk 1]
```diff
@@ -590,13 +590,33 @@ def __eq__(self, other):
...
+        unique will not be compared by default. 'id' and 'name' are examples of
+        unique fields commonly ignored. Check Entities fields for fields marked
+        with 'unique=True'
+
+
+        :param other: entity to compare
+        :param filter_fcn: callable
+        :return: boolean
+        """
+        if other is None:
```
Comment (ldjebran) [inline nailgun/entity_mixins.py]: think we should add also the case where not isinstance(other, type(self))

```python 
if other is None or not isinstance(other, type(self)):
    return False
```
what do you think?

Comment (ldjebran) [inline nailgun/entity_mixins.py]: this comment apply to  __eq__ also, as we should have an exception that say that no attr  to_json_dic, instead directly return False

**Comment by rochacbruno** [comment]
@renzon needs Cpick to 6.2.z ?

**Comment by renzon** [comment]
@rochacbruno I am not sure, once it is not used by know. But if we start to use it, Cpick would be nice to made Cpicking on robottelo easier. What you think?

**Comment by rochacbruno** [comment]
@renzon it is better to Cpick now, because in future it may end in errors if robottelo tests are implemented and then Cpicked.

**Pull request merged**